### ISSUE TITLE
feature: Add support for graphql-java-extended-validation

### ISSUE THREAD

**Issue by setchy** [Netflix/dgs-framework#640]
## Describe the Feature Request

Just like DGS supports `graphql-java-extended-scalars`, it would be nice if there was support for the suite of schema validation directives provided by [graphql-java-extended-validation](https://github.com/graphql-java/graphql-java-extended-validation)

## Describe Preferred Solution

Add support for [graphql-java-extended-validation](https://github.com/graphql-java/graphql-java-extended-validation)

**Comment by hantsy** [comment]
Note: Dgs used an old GraphQL Java 16.x, when using this `graphql-java-extended-validation`,  you have to use the 16.0 version and an older Hibernate validator, check [here](https://github.com/hantsy/spring-graphql-sample/blob/java17/dgs-codegen/build.gradle#L26-L29). 

I hope Dgs will upgrade to GraphQL Java 17 in the next minor version, there is a PR for it, https://github.com/Netflix/dgs-framework/pull/518

**Comment by paulbakker** [comment]
We're just talking about `graphql-java-extended-validation` and want to look into it, so this is very timely. We'll follow up once we have looked into it a bit more.

**Comment by paulbakker** [comment]
> Note: Dgs used an old GraphQL Java 16.x, when using this `graphql-java-extended-validation`, you have to use the 16.0 version and an older Hibernate validator, check [here](https://github.com/hantsy/spring-graphql-sample/blob/java17/dgs-codegen/build.gradle#L26-L29).
> 
> I hope Dgs will upgrade to GraphQL Java 17 in the next minor version, there is a PR for it, #518

We're still blocked by the [federation-lib](https://github.com/apollographql/federation-jvm/pull/122) release to update to 17. We raised the issue again with folks involved, so hopefully we can move on soon.

**Comment by sbilello** [comment]
For the interested readers:

I integrated `graphql-java-extended-validation` library in our project and the steps to make it work are 2:

I forced  `hibernate-validator` to solve a run time problem:

In `build.gradle`

```gradle
configurations.all {
	exclude group: 'com.vaadin.external.google' , module: 'android-json'
	resolutionStrategy {
		force 'org.hibernate.validator:hibernate-validator:6.1.6.Final'
	}
}

```

I added a new Bean

```
@DgsComponent
public class GraphQLSchemaValidationConfiguration {

  @DgsRuntimeWiring
  public RuntimeWiring.Builder addValidationDirectives(RuntimeWiring.Builder builder) {
    ValidationRules validationRules = ValidationRules.newValidationRules()
        .onValidationErrorStrategy(OnValidationErrorStrategy.RETURN_NULL)
        .build();
    ValidationSchemaWiring schemaWiring = new ValidationSchemaWiring(validationRules);
    return builder.directiveWiring(schemaWiring);
  }

}
```
I hope it helps

**Comment by berngp** [comment]
Changes are in master. The feature will be available in the next release.

**Comment by berngp** [comment]
Changes were released in [v4.9.10](https://github.com/Netflix/dgs-framework/releases/tag/v4.9.10). Please take it for a spin and let us know.

Thanks @hantsy for the feature!
